### PR TITLE
merge float div into LaneDivFP

### PR DESCRIPTION
**Pull request by midnighter95**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -3,7 +3,8 @@
 	url = git@github.com:chipsalliance/chisel.git
 [submodule "dependencies/arithmetic"]
 	path = dependencies/arithmetic
-	url = git@github.com:sequencer/arithmetic.git
+    url = git@github.com:midnighter95/arithmetic.git
+    branch = rebase
```
Comment (sequencer) [inline .gitmodules:7]: open a PR at there.

File: build.sc [hunk 2]
```diff
@@ -18,6 +18,10 @@ object v {
   val bc = ivy"org.bouncycastle:bcprov-jdk15to18:latest.integration"
   val spire = ivy"org.typelevel::spire:latest.integration"
   val evilplot = ivy"io.github.cibotech::evilplot:latest.integration"
+
+  val scalatest = ivy"org.scalatest::scalatest:3.2.0"
+  val scalapar = ivy"org.scala-lang.modules::scala-parallel-collections:1.0.4"
+  val oslib =  ivy"com.lihaoyi::os-lib:0.9.1"
```
Comment (sequencer) [inline build.sc:24]: arithmetic doesn't need this anymore.

**Review by sequencer — CHANGES_REQUESTED** [review]

File: build.sc (hunk 2 above)
Comment (midnighter95) [reply build.sc:28]: These 3 dependencies are to enable testfloat test for FP Module for now.

Comment (sequencer) [reply build.sc:28]: test should only exist arithmetic

File: .github/passed/v1024-l8-b2-fp/debug/passed.json [hunk 3]
```diff
@@ -1,3 +1,496 @@
 {
-  
-}
\ No newline at end of file
+  "rvv-vp-intrinsic-add-mlir": 343,
```
Comment (sequencer) [inline .github/passed/v1024-l8-b2-fp/debug/passed.json:2]: This file should be added by CI, init passed tests to 0.

File: .github/scripts/ci.sc [hunk 4]
```diff
@@ -250,10 +250,9 @@ def runFailedTests(jobs: String) = {
   val totalJobs = jobs.split(";")
   val failed = totalJobs.zipWithIndex.foreach { case (job, i) => {
     val Array(config, caseName, runCfg) = job.split(",")
-    val actualConfig = if (config.endsWith("-trace")) config else s"$config-trace"
-    System.err.println(s"[${i+1}/${totalJobs.length}] Running test case with trace $actualConfig,$caseName,$runCfg")
+    System.err.println(s"[${i+1}/${totalJobs.length}] Running test case with trace $config,$caseName,$runCfg")
     val handle = os
-      .proc("scripts/run-test.py", "-c", actualConfig, "-r", runCfg, "--trace", "--no-console-log", "--base-out-dir", testRunDir, caseName)
+      .proc("scripts/run-test.py", "-c", config, "-r", runCfg, "--trace", "--no-console-log", "--base-out-dir", testRunDir, caseName)
       .call(check=false)
```
Comment (sequencer) [inline .github/scripts/ci.sc:256]: I think you don't need to touch `ci.sc`?

File: nix/t1/_sources/generated.nix [hunk 5]
```diff
@@ -3,15 +3,15 @@
...
+      rev = "dd9bd585a8d444399eb5a31d088567e0ef56f43a";
       fetchSubmodules = false;
-      sha256 = "sha256-hF9sqbVtlJeACHFUF2B14CdazQfu0LTW1RM8J0tdybE=";
+      sha256 = "sha256-PquHsdyKbN/xKV4UhtTmBAVic2PwiUPvw50oZKUN08I=";
     };
-    date = "2023-11-06";
+    date = "2023-10-23";
   };
   berkeley-hardfloat = {
     pname = "berkeley-hardfloat";
```
Comment (sequencer) [inline nix/t1/_sources/generated.nix:17]: bump and use upstream

File: v/src/FloatModule.scala [hunk 6]
```diff
@@ -0,0 +1,201 @@
+package v
+
+import chisel3._
+import chisel3.util._
+import chisel3.util.experimental.decode.TruthTable
+import hardfloat._
+
+class FloatAdder(expWidth: Int, sigWidth:Int) extends Module{
```
Comment (sequencer) [inline v/src/FloatModule.scala:8]: Need reformat
```suggestion
class FloatAdder(expWidth: Int, sigWidth:Int) extends Module {
```

File: v/src/FloatModule.scala [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+    4, 3, 3, 2, 2, 1, 1, 0,
+    127, 125, 123, 121, 119, 118, 116, 114,
+    113, 111, 109, 108, 106, 105, 103, 102,
+    100, 99, 97, 96, 95, 93, 92, 91,
+    90, 88, 87, 86, 85, 84, 83, 82,
+    80, 79, 78, 77, 76, 75, 74, 73,
+    72, 71, 70, 70, 69, 68, 67, 66,
+    65, 64, 63, 63, 62, 61, 60, 59,
+    59, 58, 57, 56, 56, 55, 54, 53),7,7)
+
```
Comment (sequencer) [inline v/src/FloatModule.scala:201]: Need algorithm to get this table.

File: v/src/Lane.scala [hunk 8]
```diff
@@ -410,6 +410,8 @@ class Lane(val parameter: LaneParameter) extends Module with SerializableModule[
 
   val VFUNotClear:           Bool = Wire(Bool())
 
+  val slot0EnqueueFire: Bool = Wire(Bool())
```
Comment (sequencer) [inline v/src/Lane.scala:413]: Is this required? @qinjun-li

**Review by sequencer — CHANGES_REQUESTED** [review]

File: v/src/Lane.scala (hunk 8 above)
Comment (qinjun-li) [reply v/src/Lane.scala:413]: The first reduce data group needs to give `reduceResult` register an initial value, Slot0EnqueueFire is needed to mark whether the incoming group is the first group.

**Review by sequencer — APPROVED** [review]

**Pull request merged**